openshift_performance/ci/scripts/large_object_list/large_object_list.sh

```diff
@@ -0,0 +1,176 @@
+cycle_start_time=`date +%s`
+create_namespaces_and_secrets
+sleep 5
+
+echo "$(date) - Namespace and secret creation complete."
+echo -e "\nNamespace and secret creation complete." >> object-list-test.log &
+cycle_end_time=`date +%s` 
+total_cycle_time=$((cycle_end_time - cycle_start_time))
+echo "Total time for create cycle: $total_cycle_time s."
+
+
+# Pause to allow the system to catch up
+echo -e "\nSleeping for 5 minutes..."
+echo -e "\nSleeping for 5 minutes..." >> object-list-test.log &
+sleep 5m
+
+echo -e "\nChecking for the namespaces and secrets."
+echo -e "\nChecking for the namespaces and secrets." >> object-list-test.log &
+
+
+# Step 4. Count the secrets and namespaces, then compare the actual and expected values.
+# Find the secrets created by this test and print the output to the log file
+my_namespace_command='oc get secrets -A -l test=listobject &>> object-list-test.log &'
+eval $my_namespace_command
+
+
+# Find and count all of the secrets created by this test
+check_secrets_command="oc get secrets -A -l test=listobject --no-headers | wc -l"
+check_namespaces_command='oc get namespaces --no-headers=true -o custom-columns=:metadata.name | grep ${my_namespace} | wc -l'
+
+actual_secrets=$(eval $check_secrets_command)
+actual_namespaces=$(eval $check_namespaces_command)
+
+echo "expected # of namespaces: $expected_namespaces" 
+echo "actual # of namespaces: $actual_namespaces"
+echo "expected # of secrets: $expected_secrets"
+echo "actual # of secrets: $actual_secrets"
+
+echo "Check for the namespaces and secrets - COMPLETE"
+echo -e "\nCheck for the namespaces and secrets - COMPLETE" >> object-list-test.log &
+
+
+# Find and count operators and nodes in a bad state
+bad_operators=$(oc get co --no-headers| grep -v 'True.*False.*False' | wc -l)
+nodes_not_ready=$(oc get nodes --no-headers | grep -v Ready | wc -l)
+
+get_operator_and_node_status
+
+
+# Step 5. Print the test results
+echo ""
+echo "======Final test result======"
+if [[ ( $expected_secrets -eq $actual_secrets ) && ($expected_namespaces -eq $actual_namespaces)  && ($bad_operators -eq 0) && ($nodes_not_ready -eq 0) ]]; then
+    echo "Test passed"
+    echo "The expected and actual number of namespaces are equal."
+    echo "The expected and actual number of secrets are equal"
+    echo "Cluster operators are stable."
+    echo "All nodes are Ready."
+    echo "Removing test objects..."
+    delete_namespaces $expected_namespaces
+    exit 0
+else
+    echo "Test failed"
+    echo "Operators (oc get co --no-headers| grep -v 'True.*False.*False'):"
+    oc get co --no-headers| grep -v 'True.*False.*False'
+    echo ""
+    echo "Nodes oc get nodes --no-headers | grep -v Ready):"
+    oc get nodes --no-headers | grep -v Ready
+    echo ""
+    echo "expected # of namespaces: $expected_namespaces.  actual # of namespaces: $actual_namespaces"
+    echo "expected # of secrets: $expected_secrets.  actual # of secrets: $actual_secrets"
```

Comment:
If the number is not as expected, we want to check the exact output of the get command, better to with ' -v9' to get debug info
oc get secrets -A -l test=listobject --no-headers -v9

Reply:
And we can point to the output file location in the result section